evalml/tests/integration_tests/test_nullable_types.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+from evalml.automl import AutoMLSearch
+from evalml.pipelines.components.transformers import ReplaceNullableTypes
+from evalml.problem_types import ProblemTypes, is_time_series
+
+
+@pytest.mark.parametrize("input_type", ["pd", "ww"])
+@pytest.mark.parametrize("automl_algorithm", ["iterative", "default"])
+@pytest.mark.parametrize("problem_type", ProblemTypes.all_problem_types)
+@pytest.mark.parametrize(
+    "test_description, column_names",
+    [
+        (
+            "all null",
+            ["dates", "all_null"],
+        ),  # Should result only in Drop Null Columns Transformer
+        ("only null int", ["int_null"]),
+        ("only null bool", ["bool_null"]),
+        ("only null age", ["age_null"]),
+        ("nullable types", ["numerical", "int_null", "bool_null", "age_null"]),
+        ("just nullable target", ["dates", "numerical"]),
+    ],
+)
+def test_nullable_types_builds_pipelines(
+    automl_algorithm,
+    problem_type,
+    input_type,
+    test_description,
+    column_names,
+    get_test_data_from_configuration,
+):
+    parameters = {}
+    if is_time_series(problem_type):
+        parameters = {
+            "date_index": "dates",
+            "gap": 1,
+            "max_delay": 1,
+            "forecast_horizon": 3,
+        }
+
+    X, y = get_test_data_from_configuration(
+        input_type,
+        problem_type,
+        column_names=column_names,
+        nullable_target=True if "nullable target" in test_description else False,
+    )
+
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type=problem_type,
+        problem_configuration=parameters,
+        _automl_algorithm=automl_algorithm,
+    )
+    if automl_algorithm == "iterative":
+        pipelines = [pl.name for pl in automl.allowed_pipelines]
+    elif automl_algorithm == "default":
+        pipelines = [pl.name for pl in automl._automl_algorithm.next_batch()]
```

Comment:
Line 59 would only test the for the first batch of pipelines in `DefaultAlgorithm` and not all the pipelines that would be generated. To get the full list of pipelines you would need to follow the pattern [here](https://github.com/alteryx/evalml/blob/main/evalml/tests/automl_tests/test_default_algorithm.py#L221).  However, I don't see why behavior would change from the first batch to the next re adding the component since everything is built from `make_pipelines()`.